core/controllers/topic_editor.py

```diff
@@ -149,8 +150,49 @@ def get(self, topic_id: str) -> None:
         additional_story_summary_dicts = [
             summary.to_dict() for summary in additional_story_summaries]
 
+        canonical_stories_ids = [summary['id'] for
+            summary in canonical_story_summary_dicts]
+        canonical_stories = list(filter(
+            None, story_fetchers.get_stories_by_ids(canonical_stories_ids)))
+        canonical_stories_dict: Dict[str, story_domain.Story] = {
+            canonical_story.id: canonical_story for canonical_story in
+            canonical_stories}
         updated_canonical_story_summary_dicts = []
+
         for summary in canonical_story_summary_dicts:
+            if summary['id'] not in canonical_stories_dict:
+                continue
+            story = canonical_stories_dict[summary['id']]
+            nodes = story.story_contents.nodes
+            total_chapters_count = len(nodes)
+            published_chapters_count = 0
+            upcoming_chapters_count = 0
+            overdue_chapters_count = 0
+            upcoming_chapters_expected_days = []
+            for node in nodes:
+                if node.status == constants.STORY_NODE_STATUS_PUBLISHED:
+                    published_chapters_count += 1
+                if (node.status != constants.STORY_NODE_STATUS_PUBLISHED and
+                    node.planned_publication_date is not None):
+                    current_time_msecs = (
+                        datetime.datetime.utcnow().timestamp() * 1000)
```

Comment:
Use `get_current_time_in_millisecs()` in `utils.py` everywhere. This would also make it easy for you to mock this function in all your tests.

Reply:
Done. So it after all boils down to what I had done in the very beginning in this PR :)